### PR TITLE
Updated at on connect

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Ensures `connect`ed fields correctly have their `updatedAt` fields set when updated. Closes https://github.com/prisma/prisma/issues/4146. Thanks!

### CODE REVIEW

File: libs/prisma-models/src/datamodel_converter.rs [hunk 1]
```diff
@@ -210,24 +210,24 @@ impl<'a> DatamodelConverter<'a> {
...
-                                panic!("It's not allowed that both sides of a relation specify the inline policy. The field was {} on model {}. The related field was {} on model {}.", field.name, model.name, related_field.name, related_model.name)
-                            }
-                        },
-                    };
+                        }
+                        (Some(_), Some(_)) => {
+                            panic!("It's not allowed that both sides of a relation specify the inline policy. The field was {} on model {}. The related field was {} on model {}.", field.name, model.name, related_field.name, related_model.name)
+                        }
+                    },
+                };
```
Comment (yoshuawuyts) [inline libs/prisma-models/src/datamodel_converter.rs:230]: Not exactly sure what happened here, but CI was complaining about `cargo fmt` being off. Ran it, and this was the resulting diff.

File: query-engine/connector-test-kit/src/test/scala/queries/regressions/Prisma_4146Spec.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+       |   id           Int      @id
+       |   name         String
+       |   account      Account? @relation(fields: [accountId], references: [id])
+       |   accountId    Int?
+       |   updatedAt    DateTime @updatedAt
+       | }
+     """.stripMargin
+  }
+
+  "Updating a list of fields over a connect bound" should "change the update fields tagged with @updatedAt" taggedAs (IgnoreMsSql) in {
```
Comment (dpetrick) [inline query-engine/connector-test-kit/src/test/scala/queries/regressions/Prisma_4146Spec.scala:32]: Try with MSSQL.

**Review by dpetrick — APPROVED** [review]

File: query-engine/connector-test-kit/src/test/scala/queries/regressions/Prisma_4146Spec.scala (hunk 2 above)
Comment (yoshuawuyts) [reply query-engine/connector-test-kit/src/test/scala/queries/regressions/Prisma_4146Spec.scala:32]: good point; done!

**Comment by yoshuawuyts** [comment]
Tests pass; merging!

**Pull request merged**